src/main/java/chessmaster/game/ChessBoard.java

```diff
@@ -77,6 +79,36 @@ public void displayAvailableMoves() {
         }
     }
 
+    public void showAvailableMoves(ChessPiece piece) throws NullPieceException {
+        if (piece.isEmptyPiece()) {
+            throw new NullPieceException();
+        }
+
+        Coordinate[] coordArray = piece.getFlattenedCoordinates(this);
+
+        TextUI.printChessBoardHeader();
+        TextUI.printChessBoardDivider();
+        for (int i = 0; i < board.length; i++) {
+            ChessTile[] row = board[i];
+            StringBuilder rowString = new StringBuilder();
+
+            for (int j = 0; j < board.length; j++) {
+                Coordinate coord = new Coordinate(j, i);
+
+                if (Arrays.asList(coordArray).contains(coord)){
+                    rowString.append(row[j].toStringAvailable());
```

Comment:
Minor change for readability, cast `row[j].toStringAvailable()` into another variable
```java
String validMoveString = row[j].toStringAvailable()
rowString.append(validMoveString)
```